src/index.d.ts

```diff
@@ -0,0 +1,229 @@
+import type { FunctionComponent, ReactElement } from 'react';
+import type { AutocompleteProps } from '@mui/material/Autocomplete/Autocomplete';
+import type {
+    ButtonProps,
+    RadioGroupProps,
+    SxProps,
+    TextFieldProps,
+} from '@mui/material';
+
+/**
+ * Section to export generated type declarations of .ts or .tsx files
+ */
+
+//TODO FM to remove (for the review)
+export { useTest } from './hooks/useTest';
+export { Test } from './utils/Test';
+
+export { useIntlRef } from './hooks/useIntlRef';
+
+/**
+ * Section to export manual type declarations of .js and .jsx files
+ */
+
+export const TopBar: FunctionComponent;
+
+export function logout(dispatch: any, userManagerInstance: any): Promise<any | undefined>;
+
+export const DARK_THEME: String, LIGHT_THEME: String;
+
+interface SnackInputs {
+    messageTxt?: string;
+    messageId?: string;
+    messageValues?: { [key: string]: string };
+    headerTxt?: string;
+    headerId?: string;
+    headerValues?: Record<string, string>;
+}
+
+interface UseSnackMessageReturn {
+    snackError: (snackInputs: SnackInputs) => void;
+    snackWarning: (snackInputs: SnackInputs) => void;
+    snackInfo: (snackInputs: SnackInputs) => void;
+}
+
+export function useSnackMessage(): UseSnackMessageReturn;
+
+type Input = string | number;
+type Options = Array<{
+    id: string;
+    label: string;
+}>;
+
+interface AutocompleteInputProps
+    extends Omit<
+        AutocompleteProps<
+            string | { id: string; label: string },
+            boolean | undefined,
+            boolean | undefined,
+            boolean | undefined
+        >,
+        // we already defined them in our custom Autocomplete
+        'value' | 'onChange' | 'renderInput'
```

Comment:
We should be very careful with the Omit because it doesn't prevent to pass an object that contains these properties, and in all our component we put the remaining props at the end {...props} so if it is present it will override our values...

Reply:
It does.

![image](https://github.com/gridsuite/commons-ui/assets/75525996/82a63f79-e3c1-42be-b203-93ad0a765993)

Or I missed something ? 

This screenshot if from gridstudy (main branch), but maybe it does not works correctly with the new branch. How did you get this result ?

Reply:
yes if you do like that but if you do {...props} with an object containing value it does work , no ?

Reply:
![Capture d’écran de 2024-02-23 13-02-55](https://github.com/gridsuite/commons-ui/assets/53527338/beaf11b0-3b45-4847-9bb5-ebd7f5a2c88d)

Reply:
Ah yes weird...

Reply:
What is the correct solution then ?